Failing to create entity manager when running an app using module path

I'm getting the following exception when trying to create an entity manager when running an app using module path, instead of using classpath:

```
Exception in thread "main" jakarta.persistence.PersistenceException: java.lang.NullPointerException
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.deploy(EntityManagerSetupImpl.java:861)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.getAbstractSession(EntityManagerFactoryDelegate.java:221)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:328)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:343)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:306)
	at com.example.modpathtest@1.0.0-SNAPSHOT/com.example.modpathtest.Main.main(Main.java:10)
Caused by: java.lang.NullPointerException
	at java.base/java.io.Reader.<init>(Reader.java:168)
	at java.base/java.io.InputStreamReader.<init>(InputStreamReader.java:76)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.helper.DBPlatformHelper.load(DBPlatformHelper.java:163)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.helper.DBPlatformHelper.loadFromResource(DBPlatformHelper.java:145)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.helper.DBPlatformHelper.initializeNameToVendorPlatform(DBPlatformHelper.java:90)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.helper.DBPlatformHelper.getDBPlatform(DBPlatformHelper.java:61)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.sessions.DatabaseSessionImpl.setOrDetectDatasource(DatabaseSessionImpl.java:265)
	at org.eclipse.persistence.core@4.0.0-M3/org.eclipse.persistence.internal.sessions.DatabaseSessionImpl.loginAndDetectDatasource(DatabaseSessionImpl.java:808)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerFactoryProvider.login(EntityManagerFactoryProvider.java:259)
	at org.eclipse.persistence.jpa@4.0.0-M3/org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl.deploy(EntityManagerSetupImpl.java:772)
```

Here is a test app, which can be run with 'mvn exec:exec':

[eclipselink-module-path-test.zip](https://github.com/eclipse-ee4j/eclipselink/files/8639109/eclipselink-module-path-test.zip)

I think the problem is that DBPlatformHelper is trying to read 'org/eclipse/persistence/internal/helper/VendorNameToPlatformMapping.properties' using ClassLoader.getResourceAsStream but since package 'org.eclipse.persistence.internal.helper' is not opened unconditionally, getResourceAsStream returns null.

From getResourceAsStream javadoc:
"... this method will only find resources in packages of named modules when the package is opened unconditionally."

I guess, a quick fix for this would be to add:

opens org.eclipse.persistence.internal.helper;

to org.eclipse.persistence.core's module-info.java

--- a/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/DBPlatformHelper.java
+++ b/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/DBPlatformHelper.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1998, 2021 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1998, 2022 Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 1998, 2021 IBM Corporation. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
@@ -87,9 +87,8 @@ public class DBPlatformHelper {
             if(_nameToVendorPlatform == null) {
                 _nameToVendorPlatform = new ArrayList<>();
                 try {
-                    loadFromResource(_nameToVendorPlatform, VENDOR_NAME_TO_PLATFORM_RESOURCE_NAME,
-                                            DBPlatformHelper.class.getClassLoader() );
-                } catch (IOException e) {
+                    loadFromResource(_nameToVendorPlatform, VENDOR_NAME_TO_PLATFORM_RESOURCE_NAME);
+                } catch (Throwable e) {
                     logger.log(SessionLog.WARNING, SessionLog.CONNECTION, "dbPlatformHelper_noMappingFound", VENDOR_NAME_TO_PLATFORM_RESOURCE_NAME);
                 }
             }
@@ -140,9 +139,9 @@ public class DBPlatformHelper {
     }
 
     //-----Property Loading helper methods ----/
-    private static void loadFromResource(List<String[]> properties, String resourceName, ClassLoader classLoader)
+    private static void loadFromResource(List<String[]> properties, String resourceName)
             throws IOException {
-        load(properties, resourceName, classLoader);
+        load(properties, resourceName);
     }
 
     /**
@@ -153,14 +152,11 @@ public class DBPlatformHelper {
      *                      If loadFromFile  is true, this is fully qualified path name to a file.
      *                      param classLoader is ignored.
      *                      If loadFromFile  is false,this is resource name.
-     * @param classLoader   The class loader that should be used to load the resource. If null,primordial
-     *                      class loader is used.
      */
-    private static void load(List<String[]> properties, final String resourceName,
-            final ClassLoader classLoader)
+    private static void load(List<String[]> properties, final String resourceName)
                             throws IOException {
         try (BufferedReader bin = new BufferedReader(
-                new InputStreamReader(openResourceInputStream(resourceName,classLoader)))) {
+                new InputStreamReader(openResourceInputStream(resourceName)))) {
             for (String line = bin.readLine(); line != null; line = bin.readLine()) {
                 String[] keyValue = validateLineForReturnAsKeyValueArray(line);
                 if (keyValue != null) {
@@ -173,16 +169,10 @@ public class DBPlatformHelper {
     /**
      * Open resourceName as input stream inside doPriviledged block
      */
-    private static InputStream openResourceInputStream(final String resourceName, final ClassLoader classLoader) {
-        return PrivilegedAccessHelper.callDoPrivileged(
-                () -> {
-                    if (classLoader != null) {
-                        return classLoader.getResourceAsStream(resourceName);
-                    } else {
-                        return ClassLoader.getSystemResourceAsStream(resourceName);
-                    }
-                }
-        );
+    private static InputStream openResourceInputStream(final String resourceName) throws IOException {
+        return PrivilegedAccessHelper.callDoPrivilegedWithException(
+                () -> DBPlatformHelper.class.getModule().getResourceAsStream(resourceName),
+                (ex) -> (IOException) ex);
     }
 
     private static String[] validateLineForReturnAsKeyValueArray(String line) {

